lib/filters/core-filters.js

```diff
@@ -2,14 +2,39 @@
 const { runWhenEnvIsAvailable, external } = require('./api')
 const { addFilter, getFilter } = external
 
-/**
- * Logs an object in the template to the console in the browser.
- * @param  {Any} a any type
- * @return {String} a script tag with a console.log call.
- * @example {{ "hello world" | log }}
- * @example {{ "hello world" | log | safe }}  [for environments with autoescaping turned on]
- */
 runWhenEnvIsAvailable(() => {
   const nunjucksSafe = getFilter('safe')
+
+  /**
+   * Logs an object in the template to the console in the browser.
+   * @param  {Any} a any type
+   * @return {String} a script tag with a console.log call.
+   * @example {{ "hello world" | log }}
+   * @example {{ "hello world" | log | safe }}  [for environments with autoescaping turned on]
+   */
   addFilter('log', a => nunjucksSafe('<script>console.log(' + JSON.stringify(a, null, '\t') + ');</script>'))
+
+  /**
+   * Returns an array of objects for use in a macro that requires a list of items
+   * @param  {String[]} an array of strings.
```

Comment:
Just tried this in Visual Studio Code to see what was output

We'll want to add the missing parameter name `items`

```suggestion
   * @param  {string[]} items - an array of strings.
```

<br>

But the main thing was the JSDoc block being above `addFilter` rather than the function param

Something like this works. Typically with ~`String[]`~ `string[]` when `new String()` isn't used

```mjs
  addFilter(
    'formatItems',
  
    /**
     * Returns an array of objects for use in a macro that requires a list of items
     *
     * @param {string[]} items - an array of strings.
     * @returns {object[]} an array of objects with each object containing text and value properties.
     * @example
     * ```njk
     * {{ govukCheckboxes({
     *   name: "waste",
     *   fieldset: {
     *     legend: {
     *       text: "Which types of waste do you transport?",
     *       isPageHeading: true,
     *       classes: "govuk-fieldset__legend--l"
     *     }
     *   },
     *   hint: {
     *     text: "Select all that apply."
     *   },
     *   items: ["Rubble", "Oil", "Card"] | formatItems
     * }) }}
     * ```
     */
    (items = []) => {
      if (!Array.isArray(items)) {
        return []
      }
  
      return items
        .filter((item) => typeof item === 'string')
        .map((item) => ({ text: item, value: item }))
    }
  )
```

<img width="579" alt="Tooltip example" src="https://github.com/alphagov/govuk-prototype-kit/assets/415517/2eb6eb2b-ba18-4c90-8cb1-b0523e724fe3">

Reply:
Sorry, should have said, I've added a string filter too

Otherwise we could be assigning `text` and `value` to anything really